impress-org/give-convertkit
# Refactor: convertkit has block requirements

Resolves [GIVE-454]

## Description
- Blocks/Fields like Converkit have requirements to work.  They rely on being connected to an external service - in this case using the mailchimp api.  These are things we need to check before trying to load the block in the form builder and also the field in the donation form.

This PR uses a new method `validateApiCredentials` to determine whether to show the block or not. In the builder if it is not valid an invalid placeholder as well as a notice in the inspector will display. Both the placeholder & the notice have links that will direct users to the Settings page to complete setup for the block.

As Converkit does not have an api key validation route I have added a new setting for users to include their api secret. With the api secret & the api key we can validate the existence of the users account.

## Additional context
- Replaced the window declaration with a separate file.
- Updated styles for the template and field.
- Additional log statement for successful subscription.
- Update default checked to checked in template.

## Affects
VFB
Convertkit block

## Visuals

https://github.com/impress-org/give-convertkit/assets/75056371/5e266083-eb43-4df1-bf5b-9426983c1cd6

## Testing Instructions
- Activate Convertkit 
- Try adding the Convertkit block without setting up the apikey, it should display the invalid block placeholder & notice.
- If you have Enable globally set from the settings page the block is usually is added to any new form. If no apikey is available the block should not be added.
- Test that the links on the placeholder and the invalid block direct users to the correct settings page.
- Setup the Convertkit settings by adding your key, the block should now be able to be added as normal for both existing and new forms.

## Pre-review Checklist




[GIVE-239]: https://stellarwp.atlassian.net/browse/GIVE-239?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-446]: https://stellarwp.atlassian.net/browse/GIVE-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-448]: https://stellarwp.atlassian.net/browse/GIVE-448?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-454]: https://stellarwp.atlassian.net/browse/GIVE-454?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
